tools/logger/convert.c

```diff
@@ -346,7 +347,8 @@ static inline void print_table_header(void)
 			timestamp_width(global_config->time_precision);
 		snprintf(time_fmt, sizeof(time_fmt), "%%-%ds(us)%%%ds  ",
 			 ts_width, ts_width);
-		fprintf(out_fd, time_fmt, " TIMESTAMP", "DELTA");
+		sprintf(time_fmt, "%s", time_fmt, " TIMESTAMP", "DELTA");
+		fprintf(out_fd, time_fmt);
```

Comment:
This is explicitly bad -- what if we end up with a format specifier inside "time_fmt"? Also unsafe sprintf without length specifier AND having a string as both input and output.

Reply:
Yes here specifically the new code is the wrong one -- time_fmt is supposed to be a format specifier (computed on line 347 of the original code). You are NOT supposed to escape it here, because it's not accidental but intentional.

Further, your new line 351 passes a format string without arguments (time_fmt will have format specifiers), which INTRODUCES the very vulnerability you're mentioning.

Reply:
@paulstelian97 is correct, this change will break the (deprecated) sof-logger. Good catch thanks.

Static analyzers are not always right.

Reply:
`time_fmt` is (safely until proven otherwise) constructed at the line before but Klocwork is not capable of analyzing that. In this case it's only capable of a dumb check. This is a classic case of a false positive.